### PR TITLE
Feature/post filter submit functionality

### PR DESCRIPTION
**Pull request by saminahbab**
### What

- Added POST /filters/{id}/submit functionality 
- Added Kafka Producer for create export event 
- Changed component test structure to add Kafka Consumer and added component tests
- Added Avro schema for create start event
- Changed Config to add Kafka vars

### How to review

Run `make test` and `make test-component`

### Who can review

Team B

### CODE REVIEW

File: api/filters.go [hunk 1]
```diff
@@ -124,6 +124,118 @@ func (api *API) createFilter(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	filter, err := api.store.GetFilter(ctx, filterID)
+	if err != nil {
+		// error 500
+		api.respond.Error(
+			ctx,
+			w,
+			//			statusCode(err),
+			500,
```
Comment (PauloPortugal) [inline api/filters.go:142]: I would use the httpStatus 500 constant here.

File: api/filters.go [hunk 2]
```diff
@@ -124,6 +124,118 @@ func (api *API) createFilter(w http.ResponseWriter, r *http.Request) {
...
+	resp := UpdateFilterResponse{
+		model.JobState{
+			InstanceID:       filter.InstanceID,
+			FilterID:         filterID,
+			DimensionListUrl: "",
+			Events: []model.Event{
+				{
+					Timestamp: timeNow,
+					// TODO: right?
+					Name: "cantabular-export-start",
```
Comment (PauloPortugal) [inline api/filters.go:221]: @franmoore05 could you provide a bit more insight here.

File: service/kafka.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	pConfig := &kafka.ProducerConfig{
+		BrokerAddrs: cfg.KafkaConfig.Addr,
+		// TODO: Right default?
+		Topic:             cfg.KafkaConfig.ExportStartTopic,
+		MinBrokersHealthy: &cfg.KafkaConfig.ProducerMinBrokersHealthy,
+		KafkaVersion:      &cfg.KafkaConfig.Version,
+		MaxMessageBytes:   &cfg.KafkaConfig.MaxBytes,
+	}
+	// TODOwhat should the types be here really ?
+	// if cfg.KafkaConfig.SecProtocol == config.KafkaConfig.TLSProtocolFlag {
```
Comment (PauloPortugal) [inline service/kafka.go:21]: Please remove the comments.

File: api/filters.go [hunk 4]
```diff
@@ -124,6 +124,118 @@ func (api *API) createFilter(w http.ResponseWriter, r *http.Request) {
...
+
+	/*
+	   Note that this respose is different to swagger
+	   as discussed with Fran
+	*/
+	resp := UpdateFilterResponse{
+		model.JobState{
+			InstanceID:       filter.InstanceID,
+			FilterID:         filterID,
+			DimensionListUrl: "",
```
[the remark this answers is not in the document]
Comment (saminahbab) [reply api/filters.go]: added in the format shown by the create method

File: api/filters.go (hunk 1 above)
Comment (saminahbab) [reply api/filters.go]: used constants

File: service/kafka.go (hunk 3 above)
Comment (saminahbab) [reply service/kafka.go]: removed

File: api/filters.go (hunk 2 above)
Comment (saminahbab) [reply api/filters.go]: @PauloPortugal have spoken to Fran, and this is the right addition

File: api/filters.go [hunk 5]
```diff
@@ -124,6 +124,116 @@ func (api *API) createFilter(w http.ResponseWriter, r *http.Request) {
...
+		Version:    datasetVersion,
+		Filter_ID:  filterID,
+	}
+	// send the export event through Kafka
+	if err := api.ProduceCSVCreateEvent(&exportEvent); err != nil {
+
+		api.respond.Error(
+			ctx,
+			w,
+			500,
```
Comment (PauloPortugal) [inline api/filters.go:195]: it would be goo to use the `http.StatusInternalServerError` here as well.

File: api/filters.go [hunk 6]
```diff
@@ -124,6 +124,116 @@ func (api *API) createFilter(w http.ResponseWriter, r *http.Request) {
...
+		"filter_id": filterID,
+	}
+
+	filter, err := api.store.GetFilter(ctx, filterID)
+	if err != nil {
+		// error 500
+		api.respond.Error(
+			ctx,
+			w,
+			//			statusCode(err),
```
Comment (PauloPortugal) [inline api/filters.go:141]: can we remove this comment?

File: service/kafka.go [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+	"github.com/ONSdigital/dp-cantabular-filter-flex-api/config"
+	kafka "github.com/ONSdigital/dp-kafka/v3"
+)
+
+// GetKafkaProducer creates a Kafka producer. Currently just for POST filters/{id}/submit
+func GetKafkaProducer(ctx context.Context, cfg *config.Config) (kafka.IProducer, error) {
+	pConfig := &kafka.ProducerConfig{
+		BrokerAddrs: cfg.KafkaConfig.Addr,
+		// TODO: Right default?
```
Comment (PauloPortugal) [inline service/kafka.go:14]: can we remove this `TODO` or do we need to speak with someone else before this ticket is done?

**Review by PauloPortugal — COMMENTED** [review]
Looks good :) 
Please address the comments :)

File: api/filters.go (hunk 5 above)
Comment (saminahbab) [reply api/filters.go]: agreed, done!

File: service/kafka.go (hunk 7 above)
Comment (saminahbab) [reply service/kafka.go]: I think this was a hangover from earlier work on this branch. I believe the defaults are right. Removed.

**Review by PauloPortugal — APPROVED** [review]
LGTM

**Pull request merged**